dhis-2/dhis-api/src/main/java/org/hisp/dhis/visualization/ReportingParams.java

```diff
@@ -0,0 +1,159 @@
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.hisp.dhis.visualization;
+
+import static org.hisp.dhis.common.DxfNamespaces.DXF_2_0;
+
+import java.io.Serializable;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlProperty;
+import com.fasterxml.jackson.dataformat.xml.annotation.JacksonXmlRootElement;
+
+/**
+ * The ReportingParams object represents the reporting parameters for a
+ * Visualization. The reporting parameters are meant to make the Visualization
+ * more generic, as it can avoid having dynamic, selectable parameters rather
+ * than static. The ReportingParams is mainly used to assist during the Report
+ * generation.
+ */
+@JacksonXmlRootElement( localName = "reportingParams", namespace = DXF_2_0 )
+public class ReportingParams
+    implements
+    Serializable
+{
+
+    /**
+     * Determines if a de-serialized file is compatible with this class.
+     */
+    private static final long serialVersionUID = 3380878533309773050L;
+
+    /**
+     * Flag to indicate if there is a reporting period set or not.
+     */
+    private boolean reportingPeriod;
+
+    /**
+     * Indicates if there is a grandparent organisation unit set or not.
+     */
+    private boolean grandParentOrganisationUnit;
+
+    /**
+     * Indicates if there is a parent organisation unit set or not.
+     */
+    private boolean parentOrganisationUnit;
+
+    /**
+     * Indicates if there is an organisation unit set or not.
+     */
+    private boolean organisationUnit;
+
+    // -------------------------------------------------------------------------
+    // Constructors
+    // -------------------------------------------------------------------------
+
+    public ReportingParams()
+    {
+    }
+
+    public ReportingParams( boolean reportingPeriod, boolean grandParentOrganisationUnit,
+        boolean parentOrganisationUnit, boolean organisationUnit )
+    {
+        this.reportingPeriod = reportingPeriod;
+        this.grandParentOrganisationUnit = grandParentOrganisationUnit;
+        this.parentOrganisationUnit = parentOrganisationUnit;
+        this.organisationUnit = organisationUnit;
+    }
+
+    // -------------------------------------------------------------------------
+    // Logic
+    // -------------------------------------------------------------------------
+
+    public boolean isSet()

```

Comment:
It can be a problem. I will review the Report implementation.

Reply:
I think this was used by the old dhis-web-reports module which was removed.

Reply:
This was being unused because of an error in my code migration. This is being used by this class:

`@Service( "org.hisp.dhis.report.ReportService" )
public class DefaultReportService`